integration-tests/sdk/auth.it.js

```diff
@@ -190,6 +191,83 @@ describe('Auth Flows', () => {
     })
   })
 
+  describe('Token Provider', () => {
+    it('should return access_token from provided tokenInfo', async () => {
+      const tokenInfo = await authClient.anonymousFlow()
+      const tokenProvider = new TokenProvider({ sdkAuth: authClient }, tokenInfo)
+
+      const accessToken = await tokenProvider.getToken()
+      expect(accessToken).toEqual(tokenInfo.access_token)
+    })
+
+    it('should refresh expired access_token', async () => {
+      const onTokenInfoRefreshedMock = jest.fn()
+      const tokenInfo = await authClient.anonymousFlow()
+      tokenInfo.expires_at = 0 // emulate expired access_token
+
+      const tokenProvider = new TokenProvider({
+        sdkAuth: authClient,
+        onTokenInfoRefreshed: onTokenInfoRefreshedMock
+      }, tokenInfo)
+
+      const accessToken = await tokenProvider.getToken()
+      expect(accessToken).not.toEqual(tokenInfo.access_token)
+      expect(onTokenInfoRefreshedMock).toHaveBeenCalledTimes(1)
+      expect(accessToken).not.toEqual(onTokenInfoRefreshedMock.mock.calls[0][0].refresh_token)
+
+      const newTokenInfo = await tokenProvider.getTokenInfo()
+      expect(newTokenInfo.expires_at).not.toEqual(0)
+      expect(newTokenInfo.access_token).not.toEqual(tokenInfo.access_token)
+      expect(newTokenInfo.refresh_token).toEqual(tokenInfo.refresh_token)
+    })
+
+    it('should retrieve automatically tokenInfo', async () => {
```

Comment:
~should retrieve automatically tokenInfo~
should automatically retrieve tokenInfo

Reply:
updated here 8a90f03